Src/Sarif.PatternMatcher.Sdk/ValidatorBase.cs

```diff
@@ -275,19 +278,28 @@ internal static string ParseValue(string value)
 
         protected HttpClient CreateHttpClient()
         {
-            var httpClientHandler = new HttpClientHandler()
+            if (httpClient == null)
             {
-                AllowAutoRedirect = true,
-                MaxAutomaticRedirections = 10,
-            };
+                lock (sync)
```

Comment:
>sync [](http://example.com/codeflow?start=22&length=4)

can we find a solution that doesn't throttle requests to a single one at a time?

did you test the performance impact of this change? this is one of those that really should be tested.

Reply:
the idea here was to prevent multiple instances even in multithreaded. so, would lock in the first request and all  the others would be the static version.

To generate a port exaustion, we would require tons of httpclients. So, movint to threadstatic would not require locking and would also fix the current issue.